diff --git a/src/shr_const_mod.F90 b/src/shr_const_mod.F90
index 8437190..bd4b687 100644
--- a/src/shr_const_mod.F90
+++ b/src/shr_const_mod.F90
@@ -17,7 +17,7 @@ MODULE shr_const_mod
    real(R8),parameter :: SHR_CONST_CDAY    = 86400.0_R8      ! sec in calendar day ~ sec
    real(R8),parameter :: SHR_CONST_SDAY    = 86164.0_R8      ! sec in siderial day ~ sec
    real(R8),parameter :: SHR_CONST_OMEGA   = 2.0_R8*SHR_CONST_PI/SHR_CONST_SDAY ! earth rot ~ rad/sec
-   real(R8),parameter :: SHR_CONST_REARTH  = 6.37122e6_R8    ! radius of earth ~ m
+   real(R8),parameter :: SHR_CONST_REARTH  = 6.371229e6_R8   ! radius of earth used by UM13 ~ m
    real(R8),parameter :: SHR_CONST_G       = 9.80616_R8      ! acceleration of gravity ~ m/s^2
 
    real(R8),parameter :: SHR_CONST_STEBOL  = 5.67e-8_R8      ! Stefan-Boltzmann constant ~ W/m^2/K^4
@@ -87,7 +87,6 @@ contains
 !-----------------------------------------------------------------------------
 
   elemental logical function shr_const_isspval(rval)
-!$omp declare simd(shr_const_isspval)
 
      real(r8), intent(in) :: rval
 
